datadog/fwprovider/resource_datadog_azure_integration.go

```diff
@@ -257,6 +296,18 @@ func (r *integrationAzureResource) updateState(ctx context.Context, state *integ
 	state.ResourceCollectionEnabled = types.BoolValue(account.GetResourceCollectionEnabled())
 	state.CspmEnabled = types.BoolValue(account.GetCspmEnabled())
 	state.CustomMetricsEnabled = types.BoolValue(account.GetCustomMetricsEnabled())
+	state.MetricsEnabled = types.BoolValue(account.GetMetricsEnabled())
+	state.MetricsEnabledDefault = types.BoolValue(account.GetMetricsEnabledDefault())
+	state.UsageMetricsEnabled = types.BoolValue(account.GetUsageMetricsEnabled())
+
+	resourceProviderConfigs := account.GetResourceProviderConfigs()
+	state.ResourceProviderConfigs = make([]*ResourceProviderConfigModel, len(resourceProviderConfigs))
+	for i, resourceProviderConfig := range resourceProviderConfigs {
+		state.ResourceProviderConfigs[i] = &ResourceProviderConfigModel{
+			Namespace:      types.StringValue(resourceProviderConfig.GetNamespace()),
+			MetricsEnabled: types.BoolValue(resourceProviderConfig.GetMetricsEnabled()),
+		}
+	}
```

Comment:
go best practices would be to do this as:
```suggestion
	state.ResourceProviderConfigs = make([]*ResourceProviderConfigModel, 0, len(resourceProviderConfigs))
	for i, resourceProviderConfig := range resourceProviderConfigs {
		state.ResourceProviderConfigs = append(state.ResourceProviderConfigs, &ResourceProviderConfigModel{
			Namespace:      types.StringValue(resourceProviderConfig.GetNamespace()),
			MetricsEnabled: types.BoolValue(resourceProviderConfig.GetMetricsEnabled()),
		}
	}
```